zarr/core.py

```diff
@@ -2353,13 +2353,18 @@ def hexdigest(self, hashname="sha1"):
 
         return checksum
 
-    def __getstate__(self):
-        return (self._store, self._path, self._read_only, self._chunk_store,
-                self._synchronizer, self._cache_metadata, self._attrs.cache,
-                self._partial_decompress, self._write_empty_chunks, self._version)
-
-    def __setstate__(self, state):
-        self.__init__(*state)
+    def __reduce__(self):
+        args = (self.store,
```

Comment:
Is there a reason for using `self.store` instead of `self._store` as used before? Same question with other lines below

Reply:
I wasn't sure which to use, but the public properties seemed more intuitive to me. Happy to change to using the private properties if there's a material difference.

Reply:
I'd stick with what is already here